### PR TITLE
Add additional data to transactions

### LINKED ISSUE #762: Transactions should include all tags from the Scope
When a transaction is captured, it should include all tags from the scope.

More importantly, the SDK isn't adding `environment` and `release` to the transaction.

### PR DESCRIPTION
**Pull request by Tyrrrz**
Closes #762

### CODE REVIEW

**Comment by Tyrrrz** [comment]
@bruno-garcia I made it so it applies scope and adds environment and release. Is that good/sufficient?

File: src/Sentry/Protocol/ITransaction.cs [hunk 1]
```diff
@@ -18,6 +18,12 @@ public interface ITransaction : ISpan, ITransactionContext, IEventLike
         // 'new' because it adds a setter
         new string Name { get; set; }
 
+        // Should this be on IEventLike?
```
Comment (bruno-garcia) [inline src/Sentry/Protocol/ITransaction.cs:21]: Yeah we should bump this so it's available on any Event (transaction or errors)

File: src/Sentry/Internal/Hub.cs [hunk 2]
```diff
@@ -125,6 +131,17 @@ public void WithScope(Action<Scope> scopeCallback)
                 transaction.Sdk.AddPackage(protocolPackageName, nameAndVersion.Version);
             }
 
+            // Release information
+            transaction.Release ??= _options.Release ?? ReleaseLocator.GetCurrent();
+
+            // Environment information
+            var foundEnvironment = EnvironmentLocator.Locate();
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Hub.cs:138]: Good for us to GA this on time but good reminder that we need to get back to this after GA.

**Review by bruno-garcia — APPROVED** [review]

File: src/Sentry/Protocol/ITransaction.cs (hunk 1 above)
Comment (Tyrrrz) [reply src/Sentry/Protocol/ITransaction.cs:21]: The problem is currently `IEventLike` also encompasses `Scope`, which does not have `Release` :/

Comment (bruno-garcia) [reply src/Sentry/Protocol/ITransaction.cs:21]: We can add it, no problem

**Pull request merged**